src/components/PostDetails.jsx

```diff
@@ -13,32 +11,81 @@ function PostDetails() {
     // Fetch the post data using postId
     fetch(`https://boolean-api-server.fly.dev/Callumhayden99/post/${postId}`)
       .then(response => response.json())
-      .then(data => setPost(data));
+      .then(data => setPost(data))
+      .catch(error => console.error("Error fetching post:", error));
 
     // Fetch the comments for the post
     fetch(`https://boolean-api-server.fly.dev/Callumhayden99/post/${postId}/comment`)
       .then(response => response.json())
-      .then(data => setComments(data));
+      .then(data => setComments(data))
+      .catch(error => console.error("Error fetching comments:", error));
   }, [postId]);
 
   if (!post) {
     return <div>Loading...</div>;
   }
 
+  // Assuming contact is found from contacts
+  const contact = contacts.find(c => c.id === post.contactId);
+  const initials = contact ? `${contact.firstName[0]}${contact.lastName[0]}` : "";
+
+  const handleCommentChange = (postID, value) => {
+    // Logic for handling comment change
+  }
+
   return (
-    <div>
-      <h2>{post.title}</h2>
+    <div className="Individual-post">
+    <div className="post-details">
+      <div className="comment-title">
+        <div className="post-initials">{initials}</div>
+        {contact ? `${contact.firstName} ${contact.lastName}` : ""}
```

Comment:
Looks like a good way to handle cases where contact is undefined or null!  
If you're using a ternary operator and wanting to have "nothing" as the outcome when the condition is not met, I think you could also use && instead (eg: {contact && `${contact.firstName} ${contact.lastName}` }